azurelinuxagent/ga/monitor.py

```diff
@@ -201,6 +210,73 @@ def collect_and_send_events(self):
         except ProtocolError as e:
             logger.error("{0}", e)
 
+    def display_top_stats(self, top_stats, header, limit=50):
+        output = io.StringIO()
+
+        output.write(header)
+        output.write("\n")
+        output.write("Top %s lines\n" % limit)
+        for index, stat in enumerate(top_stats[:limit], 1):
+            frame = stat.traceback[0]
+            # replace "/path/to/module/file.py" with "module/file.py"
+            filename = os.sep.join(frame.filename.split(os.sep)[-2:])
+            output.write("#%s: %s:%s: %.1f KiB\n"
+                  % (index, filename, frame.lineno, stat.size / 1024))
+            line = linecache.getline(frame.filename, frame.lineno).strip()
+            if line:
+                output.write("    %s\n" % line)
+
+        other = top_stats[limit:]
+        if other:
+            size = sum(stat.size for stat in other)
+            output.write("%s other: %.1f KiB\n" % (len(other), size / 1024))
+        total = sum(stat.size for stat in top_stats)
+        output.write("Total allocated size: %.1f KiB\n" % (total / 1024))
+
+        logger.info(output.getvalue())
+
+    def display_top_stats_diff(self, top_stats, header, limit=50):
+        output = io.StringIO()
+
+        output.write(header)
+        output.write("\n")
+        output.write("Top %s lines\n" % limit)
+        for index, stat in enumerate(top_stats[:limit], 1):
+            frame = stat.traceback[0]
+            # replace "/path/to/module/file.py" with "module/file.py"
+            filename = os.sep.join(frame.filename.split(os.sep)[-2:])
+            output.write("#%s: %s:%s: total=%.1f KiB, new=%.1f KiB, new blocks=%d, total blocks=%d\n"
+                  % (index, filename, frame.lineno, stat.size_diff / 1024, stat.size / 1024, stat.count_diff, stat.count))
+            line = linecache.getline(frame.filename, frame.lineno).strip()
+            if line:
+                output.write("    %s\n" % line)
+
+        other = top_stats[limit:]
+        if other:
+            size = sum(stat.size for stat in other)
+            output.write("%s other: %.1f KiB\n" % (len(other), size / 1024))
+        total = sum(stat.size for stat in top_stats)
+        output.write("Total allocated size: %.1f KiB\n" % (total / 1024))
+
+        logger.info(output.getvalue())
+
```

Comment:
95% of these two methods is identical; please refactor to coalesce them. (The only difference I can see is the output.write inside the `for index, stat in...` loop.)